docs/arch/adr-001-new-adapter-scaffolding.md

```diff
@@ -0,0 +1,53 @@
+- Updating script to generate a new adapter plugin is tricky, must edit numerous template strings in python code, difficult to test and search for all areas changes are needed. ex. `$ python create_adapter_plugins.py --sql --title-case=MyAdapter ./ myadapter`
+- Not an interactive experience, must know and pass all required arguments to the .py file 
+- stuck with defaults, or missing dbt suggested default dependencies
+  - Options to set up Github Actions
+  - flake8 by default for linting to match other adapters
+  - mypy by default or optional for stati type checking
+  - tox by default 
+  - Ability to make a docker image by default
+- Lack of docstrings, examples, comments to help users undestand what needs to be built out.
+- Lack of pointers to what offical documentation we do have.
+
+## Options
+- Pull out current script, and move it to a new repo by itself
+  - lots of template strings difficult to parse
+- use Scaffolding tool
+  - easier to maintain
+  - interactive
+  - possibly more easily testable
+  - potential scaffolding tools
+      - [`cookiecutter`](https://cookiecutter.readthedocs.io/en/1.7.2/overview.html)
+        - Uses a json file to trigger commandline interactive session and passes variables user defines or chooses between potential defaults to fill out jinja templated variables and produce a new repo
+        - allows for manipulation of json behind scense to keep interactive session from being to long
+        - provides various ways to run program, either from fork/cloning, from a reconized cookiecutter command without cloning, or from within `dbt-core` able to take advantace of bumpversioning.
+      - [`pyscaffold`](https://cookiecutter.readthedocs.io/en/1.7.2/overview.html)
+        - creates a simple scaffold of project not as flexible as cookiecutter
+      - [`yehua`](https://yehua.readthedocs.io/en/latest/)
+        - Seems to of had similar functionality to cookiecutter but hasn't been suppoted in quite some time therefore no longer works with pip install, (unusable)
+
+## Decision
+After looking at available tooling, we have decided to use `cookiecutter` parsed through .py and generated "adapter" base file from current process and pulled out any information we would need to keep, built out the `cookiecutter.json` based variables that were built by running the .py variables and put them in way to augment things like adapter name to fit into correct place, assign correct ConnectionManager classes based on certain choices, applied file sturcture and dependency list for things dbt wants to suggest be used (ex. tox, flake8,etc.), once base file structure was built out I grabbed examples of the macros that do not have a default version and provided them as a context to be looked at, built out base steps of what all other macros do in general terms to help users make any changes to default action needed, created general docstrings for classes based on offical documentation and implementation.
```

Comment:
The wording in this paragraph is a little confusing. Could we try and make it a little clearer?

Reply:
rewrote section into two parts, one for choice of cookiecutter and build out of scaffold, one for addition of examples, and stubs, could eventually had a third for testing if we like it.